alchemy-core/src/main/java/io/rtr/alchemy/models/Treatment.java

```diff
@@ -31,6 +35,13 @@ public void setDescription(String description) {
         this.description = description;
     }
 
+    private static void validateName(String name) {
```

Comment:
An alternative way is to use the `@Pattern` bean validation annotation instead so you don't have to write a custom one here.  The downside is this would only apply to the API endpoints, not in the case the library is used directly.  I also don't think you'll need the `NameException`, you can throw `ValidationException` and dropwizard should automatically throw an appropriate exception that will get mapped to a 4xx response, if not, you can easily write a mapper for it as well.  If you do decide to keep custom validation (so you can support library usage, not just API usage), then I would move this validation into a separate utility class, as this will likely be used elsewhere (override names, experiment names)

Reply:
Also, these validation annotations would be on the `Create...Request` and `Update...Request` json payloads  models.

Reply:
I would prefer to do it in `core` rather than the API, it feels strange to leave a backdoor / have inconsistency between core/API.

I implemented this using inheritance - please advise if there is a better way.

Reply:
This isn't really the way I would go about it, `Experiment` and `Treatment` don't really have a 'is-a' relationship with `NameValidation`.  I would just add the validation logic in the core class method for creating and updating these entities and have a utility class that implements the `validateName` method as a static method.  Also, the `ValidationException` I was referring to is `javax.validation.ValidationException`.

Reply:
Another note, if you're trying to treat the name validation as a mixing, then I would suggest using an interface instead with a default method, e.g. something like:
```
public interface Named {
    String getName();
    default void validateName() {
        if (!getName().matches(...)) { ... }
    }
}
```

Reply:
I like this solution, too, but I believe this project is Java 7, and default methods of interfaces were introduced in java 8 [source](http://tutorials.jenkov.com/java/interfaces.html#interface-default-methods).